The following problem occurred when testing and installing all features.
```
CMake Error at CMakeLists.txt:1330 (add_subdirectory):
  add_subdirectory given source
  "E:/duckdb/buildtrees/duckdb/src/v1.2.1-9b00c3fe97.clean/extension/httpfs"
  which is not an existing directory.

```

--- a/ports/duckdb/httpfs.patch
+++ b/ports/duckdb/httpfs.patch
@@ -1,0 +1,20 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 4496860..fdda539 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1257,7 +1257,14 @@ endforeach()
+ # Load extensions passed through cmake config var
+ foreach(EXT IN LISTS BUILD_EXTENSIONS)
+   if(NOT "${EXT}" STREQUAL "")
+-    duckdb_extension_load(${EXT})
++    if("${EXT}" STREQUAL "httpfs")
++      duckdb_extension_load(${EXT} 
++        SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/extension/httpfs
++        INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/extension/httpfs/extension/httpfs/include
++      )
++    else()
++      duckdb_extension_load(${EXT})
++    endif()
+   endif()
+ endforeach()
+ 

--- a/ports/duckdb/portfile.cmake
+++ b/ports/duckdb/portfile.cmake
@@ -7,7 +7,8 @@ vcpkg_from_github(
     PATCHES
         bigobj.patch
         unvendor_icu_and_find_dependency.patch # https://github.com/duckdb/duckdb/pull/16176 + https://github.com/duckdb/duckdb/pull/16197
-)
+        httpfs.patch
+        )
 
 # Remove vendored dependencies which are not properly namespaced
 file(REMOVE_RECURSE
@@ -16,6 +17,16 @@ file(REMOVE_RECURSE
     "${SOURCE_PATH}/third_party/snowball"
     "${SOURCE_PATH}/third_party/tpce-tool"
 )
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH DUCKDB_HTTPFS_SOURCE_PATH
+    REPO duckdb/duckdb_httpfs
+    REF 85ac4667bcb0d868199e156f8dd918b0278db7b9
+    SHA512 5790ed795d394dd1b512aac0d1f1dc5976588d93b34381cab1d78c256428f3047682c7b662b1c855d3b19a9dbf99a6c64f32152dba347c18f2a36e19bcc3c5df
+    HEAD_REF main
+)
+
+file(RENAME "${DUCKDB_HTTPFS_SOURCE_PATH}" "${SOURCE_PATH}/extension/httpfs")
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" DUCKDB_BUILD_STATIC)
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" DUCKDB_BUILD_DYNAMIC)

--- a/versions/d-/duckdb.json
+++ b/versions/d-/duckdb.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "0e68b27b978a2dff4342f1338f56a2889ea2c217",
+      "git-tree": "213b32d9a3a2f7ef1676b475d18efa7d20879888",
       "version": "1.2.1",
       "port-version": 0
     },

